pkg/api/pod/warnings.go

```diff
@@ -303,6 +303,27 @@ func warningsForPodSpecAndMeta(fieldPath *field.Path, podSpec *api.PodSpec, meta
 		return true
 	})
 
+	//Accumulate port names of all containers
+	allPortsName := map[string][]portBlock{}
+	pods.VisitContainersWithPath(podSpec, fieldPath.Child("spec"), func(c *api.Container, fldPath *field.Path) bool {
```

Comment:
With https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/753-sidecar-containers/README.md SidecarContainer enabled, should this include `initContainer` if the restart policy is always.

Reply:
There is an example in the KEP
```
kind: Pod
spec:
  initContainers:
  - name: vault-agent
    image: hashicorp/vault:1.12.1
  - name: istio-proxy
    image: istio/proxyv2:1.16.0
    args: ["proxy", "sidecar"]
    restartPolicy: Always
  containers:
  ...
```

In the new proposal, istio-proxy is a sidecar container and valut-agent is not. 

This is container's restartPolicy, not pod's restartPolicy.

Reply:
https://github.com/kubernetes/kubernetes/blob/7fe31be11fbe9b44af262d5f5cffb1e73648aa96/pkg/apis/core/pods/helpers.go#L35 

@yanfeng1992 pointed out that `VisitContainersWithPath` covers `containers`, `initContainers` and `ephemeralContainers`. You can resolve this comment. Sorry for disturb.